src/sqlparser/src/ast/mod.rs

```diff
@@ -3153,6 +3153,8 @@ impl fmt::Display for SetVariableValueSingle {
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub enum AsOf {
     ProcessTime,
+    // used by time travel
+    ProcessTimeWithInterval((String, Option<DateTimeField>)),
```

Comment:
Should we evaluate `ProcessTimeWithInterval` to `TimestampNum` at the parser? In this way, we can unify the code.

Reply:
I've also considered that once. The problem is the Interval parser is not in sqlparser crate. Should we move it into sqlparser crate?

Reply:
I see. `ProcessTimeWithInterval((String, Option<DateTimeField>)),` this enum looks a bit special because it only can represent an expression like `CURRENT() - xxx` but not another form. Can we use an expression instead, so that we can support a more general expression like `CURRENT() + xxx - yyy`

Reply:
For example, this sql `select now() - interval '1' Day + interval '2' hour;` could be evaluated to a const value.

Reply:
As discussed offline, using expr as a better idea, can be done in further PR.